openedx_external_enrollments/tasks.py

```diff
@@ -20,3 +28,21 @@ def generate_salesforce_enrollment(data, *args, **kwargs):  # pylint: disable=un
     else:
         # Calling the controller enrollment method
         enrollment_controller._post_enrollment(data)  # pylint: disable=protected-access
+
+
+@periodic_task(
+    bind=True,
+    run_every=timedelta(
+        days=settings.OEE_VIPER_API_KEYS_REFRESH_TIMEOUT_DAYS,
+    ),
+)
+def refresh_viper_api_keys(self):
+    """
+    Periodic task that handles Viper's API keys refresh request.
+    """
+
+    result_message, status_code = ViperExternalEnrollment()._refresh_api_keys()  # pylint: disable=protected-access
+    if status_code != status.HTTP_200_OK:
+        raise self.retry(exc=Exception(result_message), countdown=COUNT_DOWN_RETRY)
+
+    return result_message
```

Comment:
@ivanvgh I wonder why you return result_message, since you already save it in EnrollmentRequestLog within _refresh_api_keys().
```suggestion
    _, status_code = ViperExternalEnrollment()._refresh_api_keys()  # pylint: disable=protected-access
    if status_code != status.HTTP_200_OK:
        raise self.retry(exc=Exception(result_message), countdown=COUNT_DOWN_RETRY)
```

Reply:
we need the result_message to raise the exception so that we will know in the logs or the task status why it failed.

Reply:
My bad, yes it is necessary when raising the exception, forget about renaming the variable to '_'.